lib/nodejs/sandtable.js

```diff
@@ -97,17 +88,20 @@ function addToCache( url, data ) {
     }
 }
 
+// move to the next request by nulling out current request and forcing a processRequest
 function nextRequest() {
     currentRequest = null;
     processRequest();
 }
 
+
 function processRequest() {
     if ( requestQueue.length > 0 && currentRequest == null ) {
+    	// update ipaddress as needed
         if ( ipaddress.length <= 0 ) {
-            Init();
+            initializeIPaddresses();
         }
-
+        // return if no ipaddress is available
```

Comment:
How does it determine which IP address to use?  If I have a series of virtual network adapters, could it mistakenly try to use one of those?

Reply:
It uses os.networkInterfaces(); to get all the network interfaces (filtering out local and non-IPv4 addresses).  It attempts to forward on the first address on the list.  If the request is denied it continues through the list of ipaddresses until one makes a connection (or until it reaches the end of the list).  Once it finds one that works, it sticks with that so it should work even if you have virtual network adapters.

Reply:
Thanks.  I have no intention on running any services on virtual machines.  I just want to make sure VMs don't break anything.  A VM should not return success if it's not running a valid server, correct?

Reply:
Yeah.  That's correct.  If a VM is not running a valid server, it would fail to make the connection and the ITDG server would pass on it.

Reply:
Great, thanks.